src/main/java/dao/H2Factory/H2OrderDao.java

```diff
@@ -3,173 +3,122 @@
-                    .prepareStatement("select * "
-                            + "from customerapplication.ordertable where id=?");
-            ResultSet rs = st.executeQuery();
-            gotOrder = this.convertToOrder(rs).get(0);
-            connectionProvider.destroy();
-        } catch (SQLException ex) {
-            LOG.error(ex.getSQLState());
-            throw new DaoException("Getting data error", ex);
+    public List<Order> get(Customer customer) {
+        String query = "select * from customerapplication.ordertable"
+                + " where id=:customerid";
+        Map params = new HashMap();
+        params.put("customerid", customer.getCardNumber());
+        List<Order> orders = jdbcTemplate.query(query,
+                params, new OrderMapper());
+        for (Order order : orders) {
+            order.addProducts(this.getOrderProducts(order.getOrderId()));
         }
-        return gotOrder;
+        return orders;
     }
 
-    private List<Product> getOrderProducts(int orderId) throws SQLException, DaoException {
-        List<Product> list = new ArrayList<Product>();
-        PreparedStatement st = connectionProvider.getConnection()
-                .prepareStatement("select "
-                        + "customerapplication.product.* from "
-                        + "customerapplication.product as product"
-                        + "inner join customerapplication.order_product"
-                        + "as order_product"
-                        + "on product.id = order_product.productid"
-                        + "where order_product.productid=?");
-        st.setInt(1, orderId);
-        ResultSet rs = st.executeQuery();
-        list = productConverter.convert(rs);
-        connectionProvider.destroy();
-        return list;
+    public Order get(int id) {
+        String query = "select * "
+                + "from customerapplication.ordertable "
+                + "where id=:orderid";
+        Map params = new HashMap();
+        params.put("orderid", id);
+        Order order = (Order) jdbcTemplate.query(query, params,
+                new OrderMapper());
+        order.addProducts(this.getOrderProducts(id));
+        return order;
     }
 
-    private List<Order> convertToOrder(ResultSet rs) throws SQLException, DaoException {
-        List<Order> list = new ArrayList<Order>();
-        while (rs.next()) {
-            int tempId = rs.getInt("id");
-            Date tempDate = rs.getDate("orderdate");
-            Customer tempCustomer = customerDao
-                    .get(rs.getInt("customerid"));
-            list.add(new Order(tempCustomer, this
-                    .getOrderProducts(tempId), tempId, tempDate));
-        }
-        return list;
+    private List<Product> getOrderProducts(int orderId) {
+        String query = "select "
+                + "customerapplication.product.* from "
+                + "customerapplication.product as product"
```

Comment:
this query shouldn't work as there are no space between '... as product' and 'inner join...' Looks like exception will be thrown.

My advice always start new sql line from space.